lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -110,8 +110,8 @@ function dosomething_user_menu_alter(&$items) {
   $items['user/%'] = array(
     'page callback' => 'dosomething_user_get_view',
     'page arguments' => array(1),
-    'access callback' => 'user_access',
-    'access arguments' => array('access user profiles'),
+    'access callback' => 'user_view_access',
```

Comment:
aka from user/[id] clicking "update my profile" goes to the /edit page. but it's not the same /edit page that an admin would see